mir_eval/separation.py

```diff
@@ -66,11 +66,27 @@ def validate(reference_sources, estimated_sources):
                          '= {}'.format(reference_sources.shape,
                                        estimated_sources.shape))
 
+    if reference_sources.ndim > 3 or estimated_sources.ndim > 3:
+        raise ValueError('The number of dimensions is too high (must be less '
+                         'than 3). reference_sources.ndim = {}, '
+                         'estimated_sources.ndim '
+                         '= {}'.format(reference_sources.ndim,
+                                       estimated_sources.ndim))
+
     if reference_sources.size == 0:
         warnings.warn("reference_sources is empty, should be of size "
                       "(nsrc, nsample).  sdr, sir, sar, and perm will all "
                       "be empty np.ndarrays")
-    elif np.any(np.all(reference_sources == 0, axis=1)):
+    elif (reference_sources.ndim == 2 and
+            np.any(np.all(reference_sources == 0, axis=1))):
+        raise ValueError('All the reference sources should be non-silent (not '
+                         'all-zeros), but at least one of the reference '
+                         'sources is all 0s, which introduces ambiguity to the'
+                         ' evaluation. (Otherwise we can add infinitely many '
+                         'all-zero sources.)')
+    elif (reference_sources.ndim > 2 and
+            np.any(np.all(np.sum(reference_sources, axis=-1) == 0,
+                          axis=1))):
```

Comment:
Can you combine the above two if-statements since they produce the same Exception?  It seems to me that you are effectively checking whether the trailing dimension is all-zeros, right?  If so, I am sure we can figure out some numpy wizardry to do this all in one case.  It also looks like you are only testing ndim >= 2, but isn't ndim == 1 possible too (which then gets converted to 2D by adding an axis)?